lib/errors/DiagnosticWarning.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Basic "warning" class. Contains a message and a type. Can be emitted by tests
+ * when necessary.
+ */
+export class DiagnosticWarning {
+  /**
+   * Message explaining the warning.
+   */
+  message: string;
+  /**
+   * When the warning was created.
+   */
+  timestamp: number;
+  /**
+   * Type of warning. For example, "low-audio-level".
+   */
+  type: string;
+
+  /**
+   * Constructor for the [[DiagnosticWarning]] class. Sets the type and message
+   * of the object.
+   * @param type
+   * @param message
+   */
+  constructor(type: string, message: string) {
+    this.message = message;
+    this.timestamp = Date.now();
```

Comment:
Was there a discussion about making this a class? It seems like all we get out of this warning being a class is an automatic timestamp. Is the timestamp useful? Did we consider making this an interface?

cc @charliesantos

Reply:
If I remember correctly, I don't think we had that discussion. I can see the benefit of making this a class so it is consistent with DiagnosticsError. If we're going to make this an interface, we should do it for both. I don't think timestamp is useful in this case tbh.